libkpod/create.go

```diff
@@ -0,0 +1,823 @@
+			}
+
+			if parts[0] == "" {
+				continue
+			}
+			specgen.AddProcessEnv(parts[0], parts[1])
+		}
+	}
+
+	// Set working directory
+	// Pick it up from image config first and override if specified in CRI
+	containerCwd := "/"
+	if containerImageConfig != nil {
+		imageCwd := containerImageConfig.Config.WorkingDir
+		if imageCwd != "" {
+			containerCwd = imageCwd
+		}
+	}
+	runtimeCwd := containerConfig.WorkingDir
+	if runtimeCwd != "" {
+		containerCwd = runtimeCwd
+	}
+	specgen.SetProcessCwd(containerCwd)
+
+	// Setup user and groups
+	if linux != nil {
+		if err = setupContainerUser(&specgen, mountPoint, linux.GetSecurityContext(), containerImageConfig); err != nil {
+			return nil, err
+		}
+	}
+
+	// Set up pids limit if pids cgroup is mounted
+	_, err = cgroups.FindCgroupMountpoint("pids")
+	if err == nil {
+		specgen.SetLinuxResourcesPidsLimit(c.config.PidsLimit)
+	}
+
+	// by default, the root path is an empty string. set it now.
+	specgen.SetRootPath(mountPoint)
+
+	saveOptions := generate.ExportOptions{}
+	if err = specgen.SaveToFile(filepath.Join(containerInfo.Dir, "config.json"), saveOptions); err != nil {
+		return nil, err
+	}
+	if err = specgen.SaveToFile(filepath.Join(containerInfo.RunDir, "config.json"), saveOptions); err != nil {
+		return nil, err
+	}
+
+	container, err := oci.NewContainer(containerID, containerName, containerInfo.RunDir, logPath, sb.NetNs(), labels, kubeAnnotations, image, metadata, sb.ID(), containerConfig.Tty, containerConfig.Stdin, containerConfig.StdinOnce, sb.Privileged(), sb.Trusted(), containerInfo.Dir, created, containerImageConfig.Config.StopSignal)
+	if err != nil {
+		return nil, err
+	}
+
+	return container, nil
+}
+
+func (c *ContainerServer) setupSeccomp(specgen *generate.Generator, cname string, sbAnnotations map[string]string) error {
+	profile, ok := sbAnnotations["security.alpha.kubernetes.io/seccomp/container/"+cname]
+	if !ok {
+		profile, ok = sbAnnotations["security.alpha.kubernetes.io/seccomp/pod"]
+		if !ok {
+			// running w/o seccomp, aka unconfined
+			profile = seccompUnconfined
+		}
+	}
+	if !c.seccompEnabled {
+		if profile != seccompUnconfined {
+			return fmt.Errorf("seccomp is not enabled in your kernel, cannot run with a profile")
+		}
+		logrus.Warn("seccomp is not enabled in your kernel, running container without profile")
+	}
+	if profile == seccompUnconfined {
+		// running w/o seccomp, aka unconfined
+		specgen.Spec().Linux.Seccomp = nil
+		return nil
+	}
+	if profile == seccompRuntimeDefault {
+		return seccomp.LoadProfileFromStruct(c.seccompProfile, specgen)
+	}
+	if !strings.HasPrefix(profile, seccompLocalhostPrefix) {
+		return fmt.Errorf("unknown seccomp profile option: %q", profile)
+	}
+	//file, err := ioutil.ReadFile(filepath.Join(s.seccompProfileRoot, strings.TrimPrefix(profile, seccompLocalhostPrefix)))
+	//if err != nil {
+	//return err
+	//}
+	// TODO(runcom): setup from provided node's seccomp profile
```

Comment:
Probably yes, but I don't believe we can do it until we refactor the API to not use Kubernetes CRI structs.